comments/email_deliverability.md

```diff
@@ -0,0 +1,46 @@
+# Email deliverability
+
+**2024-09-18**
+
+SPF, DKIM, and DMARC are email authentication protocols used to verify the legitimacy of emails, reduce email spoofing, and improve email deliverability.
+
+### SPF
+Ensures that the email is being sent from a legitimate source (i.e., an authorized mail server):
+* Requirement - an SPF record in DNS is published, listing authorized IP addresses or servers.
+* Flow:
+  1. When an email is received, the receiving server checks the SPF record for the domain in the Return-Path (envelope sender).
+  2. If the sender's IP is listed in the SPF record, the email passes SPF. If not, it fails.
+
+### DKIM
+Ensures that the email content hasn’t been tampered with during transit:
+* Requirement - a public key is published in the domain's DNS.
+* Flow:
+  1. The domain owner generates a DKIM signature using a private key and attaches it to the email header.
+  2. The receiving server retrieves the public key from DNS and verifies the DKIM signature to ensure the email hasn’t been altered during transit. If the signature matches, the email passes DKIM.
+
+### DMARC
+Combines both SPF and DKIM to provide a policy framework for how receiving mail servers should handle unauthenticated emails.
+* Requirement - DMARC policy is published in DNS, specifying actions for emails that fail SPF and/or DKIM (e.g., reject, quarantine, or allow).
+* Flow:
+  1. DMARC checks if the domain in the visible From address matches the domain in the SPF/DKIM authentication.
+  2. If both SPF and DKIM pass (or at least one, depending on the policy) and they align with the From domain, the email passes DMARC:
+      * for SPF the "Return-Path" should match the "From" domain
+      * for DKIM the signature's domain (`d=`) value should match the "From" domain
+  3. If not, the receiving server follows the action specified in the DMARC policy (e.g., mark as spam, reject).
```

Comment:
Were you able to determine if DMARC policy settings themselves have any affect on message deliverability? Or is it just a matter of having *any* DMARC entry for a domain, and the problem we were/are facing is that our use of SES servers means the DMARC entry for `visual-meaning.com` is not respected?

Reply:
We already have a DMARC entry for the domain and it applies to subdomains so there is no good way to not have it. The DMARC entry itself shouldn't really matter in respect to deliverability it determines the aftereffect - what to do if not delivered. The issues we were having were insufficiently configured SPF and DKIM.